Explorer/Assets/DCL/PluginSystem/Global/MinimapPlugin.cs

```diff
@@ -8,82 +8,81 @@
 using DCL.UI.MainUI;
 using ECS;
 using ECS.SceneLifeCycle;
-using ECS.SceneLifeCycle.Realm;
 using Global.Dynamic;
 using MVC;
+using System;
 using System.Threading;
-using ECS.SceneLifeCycle;
-using ECS.SceneLifeCycle.Realm;
-using Utility.Tasks;
 
 namespace DCL.PluginSystem.Global
 {
     public class MinimapPlugin : IDCLGlobalPlugin<MinimapPlugin.MinimapSettings>
     {
         private readonly IMVCManager mvcManager;
-        private readonly MapRendererContainer mapRendererContainer;
-        private readonly IPlacesAPIService placesAPIService;
-        private readonly IRealmData realmData;
-        private readonly IRealmNavigator realmNavigator;
-        private readonly IChatMessagesBus chatMessagesBus;
-        private readonly IScenesCache scenesCache;
-        private readonly MainUIView mainUIView;
-        private readonly IMapPathEventBus mapPathEventBus;
-        private MinimapController minimapController;
-        private readonly ISceneRestrictionBusController sceneRestrictionBusController;
-
-        private readonly string startParcelInGenesis;
+        private readonly Lazy<MinimapController> lazyMap;
 
-        public MinimapPlugin(IMVCManager mvcManager, MapRendererContainer mapRendererContainer, IPlacesAPIService placesAPIService,
-            IRealmData realmData, IChatMessagesBus chatMessagesBus, IRealmNavigator realmNavigator, IScenesCache scenesCache, MainUIView mainUIView,
+        private MinimapPlugin(IMVCManager mvcManager, MapRendererContainer mapRendererContainer, IPlacesAPIService placesAPIService,
+            IRealmData realmData, IChatMessagesBus chatMessagesBus, IScenesCache scenesCache, MainUIView mainUIView,
             IMapPathEventBus mapPathEventBus, ISceneRestrictionBusController sceneRestrictionBusController,
             string startParcelInGenesis)
         {
             this.mvcManager = mvcManager;
-            this.mapRendererContainer = mapRendererContainer;
-            this.placesAPIService = placesAPIService;
-            this.realmData = realmData;
-            this.chatMessagesBus = chatMessagesBus;
-            this.realmNavigator = realmNavigator;
-            this.scenesCache = scenesCache;
-            this.mapPathEventBus = mapPathEventBus;
-            this.mainUIView = mainUIView;
-            this.sceneRestrictionBusController = sceneRestrictionBusController;
-            this.startParcelInGenesis = startParcelInGenesis;
+
+            lazyMap = new Lazy<MinimapController>(() =>

```

Comment:
Because I got rid off the event flow that is wonky with callbacks and I passed MinimapController directly to call its method

Reply:
Can you give further details? Which event flow you got rid off?

Reply:
Got it, but do you need the `LazyMap`? I did a quick test on this [branch](https://github.com/decentraland/unity-explorer/commit/438c0d96f60e3326e3763a1f98c6b81053869c87).

Initializating `MapController` on the `MinimapPlugin.cs` constructor seems to work just fine.

Reply:
Forget it, I think I found the problem you solved. 

The issue is the cyclic dependency on `container.ChatMessagesBus`. If we do it eagerly, that reference is passed as null, and the button wont work. So, I guess that the lazy initialization is fine

Reply:
@NickKhalow 

Actually, I was wrong. As it is, the button `Go Back to Genesis City` does not work. So, going back to the previous comment, the lazy initialization is not needed; and we have to fix the issue with the button

Reply:
Actually refactor hadn't been dedicated to `ChatMessagesBus`

It was a dependency conflict with `RealmMisc`, because it requires to directly call `MinimapController` and I wanted to avoid moving `MinimapController` instantiation to `DynamicWorldContainer`

Reply:
I removed lazy map because it won't solve the original problem. I left it with `Inject()` hack but it should be redesigned with my proposal for DynamicWorldContainer.

just copy here for history:

_I noticed an issue in DynamicWorldContainer. It's possible to pass an uninitialized dependency to an object that requires it. For example, you might pass container.ChatMessagesBus, assuming it's initialized, but in reality, it's only initialized dozens of lines later. This can lead to a null reference error in a completely different part of the code due to the invalid dependency.
To solve this, I propose to make DynamicWorldContainer immutable and construct it only at the return steps. This approach will prevent such issues at compile time by ensuring that references cannot be passed before they are initialized._